### PR TITLE
DEP: deprecate trapz alias of `stats.trapezoid` distribution

### LINKED ISSUE #20486: DEP: deprecate and remove remaining usages of slur-adjacent "trapz"
I am not finding any other appearances of these terms except for `trapz`, which is still allowed as an alias for the trapezoidal distribution and as a `method` option of `quad_vec`. (Those uses don't appear to have been deprecated.)

_Originally posted by @mdhaber in https://github.com/scipy/scipy/pull/20278#pullrequestreview-1944140754_

### PR DESCRIPTION
**Pull request by h-vetinari**
Fixes #20486

Needs to go into 1.14, but can land after rc1

### CODE REVIEW

File: scipy/stats/_continuous_distns.py [hunk 1]
```diff
@@ -9683,12 +9683,41 @@ def _entropy(self, c, d):
...
+
+# since the deprecated class gets intantiated upon import (and we only want to
+# warn upon use), add the deprecation to each class method
+_method_names = [
+    "cdf", "entropy", "expect", "fit", "interval", "isf", "logcdf", "logpdf",
+    "logsf", "mean", "median", "moment", "pdf", "ppf", "rvs", "sf", "stats",
+    "std", "var"
+]
+for m in _method_names:
+    wrapper = np.deprecate(getattr(trapz, m), f"trapz.{m}", f"trapz.{m}",
```
Comment (j-bowhay) [inline scipy/stats/_continuous_distns.py:9716]: Possible copy paste error?
```suggestion
    wrapper = np.deprecate(getattr(trapz, m), f"trapz.{m}", f"trapezoid.{m}",
```

**Review by j-bowhay — COMMENTED** [review]
Thanks @h-vetinari, one query

**Comment by h-vetinari** [comment]
```
================================== FAILURES ===================================
_________________________________ test_cython _________________________________
[gw0] win32 -- Python 3.12.3 C:\hostedtoolcache\windows\Python\3.12.3\x64\python.exe
Test passed but took too long to run: Duration 22.02283800000032s > 20s
```
Why do we have such tight time bounds on this...? Sure we don't want to accidentally blow up the runtime of the test suite, but it shouldn't be so tight that we get spurious failures on randomly slower agents.

**Comment by mdhaber** [comment]
I will adjust it. The bounds weren't that tight. It is quite variable.

**Review by j-bowhay — COMMENTED** [review]
Have commented over at https://github.com/h-vetinari/scipy/pull/1

**Comment by h-vetinari** [comment]
@j-bowhay, please commit whatever fixes your approach needs directly into this PR

**Comment by h-vetinari** [comment]
Cannot approve my own PR, but this should be good now IMO. Thanks for the review/help!

**Pull request merged**